src/GT.UI/Controllers/ListingsController.cs

```diff
@@ -67,52 +78,36 @@ public async Task<IActionResult> PutListing(string id, Listing listing)
 			}
 
 			return NoContent();
+
 		}
 
 		// POST: api/Listings
 		// To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
 		[HttpPost]
-		public async Task<ActionResult<Listing>> PostListing(Listing listing)
+		public async Task<ActionResult<Listing>> PostListing(ListingDTO listing)
 		{
-			_context.Listings.Add(listing);
-			try
-			{
-				await _context.SaveChangesAsync();
-			}
-			catch (DbUpdateException)
-			{
-				if (ListingExists(listing.Id))
-				{
-					return Conflict();
-				}
-				else
-				{
-					throw;
-				}
-			}
-
+			await _ListingService.AddAsync(listing, _userManager.GetUserId(User));
 			return CreatedAtAction("GetListing", new { id = listing.Id }, listing);
 		}
 
 		// DELETE: api/Listings/5
 		[HttpDelete("{id}")]
 		public async Task<IActionResult> DeleteListing(string id)
 		{
-			var listing = await _context.Listings.FindAsync(id);
+			var listing = await _ListingService.GetByIdAsync(id);
 			if (listing == null)
 			{
 				return NotFound();
 			}
 
-			_context.Listings.Remove(listing);
-			await _context.SaveChangesAsync();
+			await _ListingService.DeleteAsync(id);
 
 			return NoContent();
 		}
 
 		private bool ListingExists(string id)
```

Comment:
Is this method necessary as we have _ListingService.ExistsById in service?

Reply:
Yupp forgot to delete it